CMakeModules/build_clBLAS.cmake

```diff
@@ -1,42 +1,49 @@
-INCLUDE(ExternalProject)
-
-SET(prefix ${PROJECT_BINARY_DIR}/third_party/clBLAS)
-SET(clBLAS_location ${prefix}/lib/import/${CMAKE_STATIC_LIBRARY_PREFIX}clBLAS${CMAKE_STATIC_LIBRARY_SUFFIX})
-IF(CMAKE_VERSION VERSION_LESS 3.2)
-    IF(CMAKE_GENERATOR MATCHES "Ninja")
-        MESSAGE(WARNING "Building clBLAS with Ninja has known issues with CMake older than 3.2")
-    endif()
-    SET(byproducts)
-ELSE()
-    SET(byproducts BUILD_BYPRODUCTS ${clBLAS_location})
-ENDIF()
+# Copyright (c) 2017, ArrayFire
+# All rights reserved.
+#
+# This file is distributed under 3-clause BSD license.
+# The complete license agreement can be obtained at:
+# http://arrayfire.com/licenses/BSD-3-Clause
+
+include(ExternalProject)
+
+set(prefix ${PROJECT_BINARY_DIR}/third_party/clBLAS)
+set(clBLAS_location ${prefix}/lib/import/${CMAKE_STATIC_LIBRARY_PREFIX}clBLAS${CMAKE_STATIC_LIBRARY_SUFFIX})
+
+find_package(OpenCL)
 
 ExternalProject_Add(
     clBLAS-ext
     GIT_REPOSITORY https://github.com/arrayfire/clBLAS.git
     GIT_TAG arrayfire-release
-    ${byproducts}
+    BUILD_BYPRODUCTS ${clBLAS_location}
     PREFIX "${prefix}"
     INSTALL_DIR "${prefix}"
     UPDATE_COMMAND ""
-    CONFIGURE_COMMAND ${CMAKE_COMMAND} -Wno-dev "-G${CMAKE_GENERATOR}" <SOURCE_DIR>/src
-    -DCMAKE_CXX_COMPILER:FILEPATH=${CMAKE_CXX_COMPILER}
-    "-DCMAKE_CXX_FLAGS:STRING=${CMAKE_CXX_FLAGS} -w -fPIC"
-    -DCMAKE_C_COMPILER:FILEPATH=${CMAKE_C_COMPILER}
-    "-DCMAKE_C_FLAGS:STRING=${CMAKE_C_FLAGS} -w -fPIC"
-    -DCMAKE_BUILD_TYPE:STRING=${CMAKE_BUILD_TYPE}
-    -DCMAKE_INSTALL_PREFIX:PATH=<INSTALL_DIR>
-    -DBUILD_SHARED_LIBS:BOOL=OFF
-    -DBUILD_CLIENT:BOOL=OFF
-    -DBUILD_TEST:BOOL=OFF
-    -DBUILD_KTEST:BOOL=OFF
-    -DSUFFIX_LIB:STRING=
+    DOWNLOAD_NO_PROGRESS 1
+    CONFIGURE_COMMAND ${CMAKE_COMMAND} -Wno-dev <SOURCE_DIR>/src
+      -DCMAKE_CXX_FLAGS:STRING="-fPIC"
+      -DCMAKE_C_FLAGS:STRING="-fPIC"
+      -DCMAKE_INSTALL_PREFIX:PATH=<INSTALL_DIR>
+      -DBUILD_SHARED_LIBS:BOOL=OFF
+      -DBUILD_CLIENT:BOOL=OFF
+      -DBUILD_TEST:BOOL=OFF
+      -DBUILD_KTEST:BOOL=OFF
+      -DSUFFIX_LIB:STRING=
+
+      # clBLAS uses a custom FindOpenCL that doesn't work well on Ubuntu
+      -DOPENCL_LIBRARIES:FILEPATH=${OpenCL_LIBRARIES}
     )
 
 ExternalProject_Get_Property(clBLAS-ext install_dir)
-ADD_LIBRARY(clBLAS IMPORTED STATIC)
-SET_TARGET_PROPERTIES(clBLAS PROPERTIES IMPORTED_LOCATION ${clBLAS_location})
-ADD_DEPENDENCIES(clBLAS clBLAS-ext)
-SET(CLBLAS_INCLUDE_DIRS ${install_dir}/include)
-SET(CLBLAS_LIBRARIES clBLAS)
-SET(CLBLAS_FOUND ON)
+
+set(CLBLAS_INCLUDE_DIRS ${install_dir}/include)
+set(CLBLAS_LIBRARIES clBLAS)
+set(CLBLAS_FOUND ON)
+make_directory("${CLBLAS_INCLUDE_DIRS}")
+
+add_library(clBLAS UNKNOWN IMPORTED)
+set_target_properties(clBLAS PROPERTIES
+  IMPORTED_LOCATION "${clBLAS_location}"
+  INTERFACE_INCLUDE_DIRECTORIES "${CLBLAS_INCLUDE_DIRS}")
+add_dependencies(clBLAS clBLAS-ext)
```

Comment:
Can we have consistent library importing mechanism, most of graphics and imaging related dependencies use naming scheme such as 'FreeImage::FreeImage'. Why are we not using similar naming convention here?

Reply:
Addressed